p2p/server.go

```diff
@@ -876,6 +885,22 @@ func (srv *Server) listenLoop() {
 	}
 }
 
+func (srv *Server) checkInboundConn(fd net.Conn, remoteIP net.IP) error {
+	if remoteIP != nil {
+		// Reject connections that do not match NetRestrict.
+		if srv.NetRestrict != nil && !srv.NetRestrict.Contains(remoteIP) {
+			return fmt.Errorf("not whitelisted in NetRestrict")
+		}
+		// Reject Internet peers that try too often.
+		srv.inboundHistory.expire(time.Now())
```

Comment:
I am not 100% clear on the whole dial/server implementation, but given that the listenLoop is running in its own goroutine, isn't this direct call to expire at risk from parallel modifications to the heap from elsewhere, because heap implementation is not thread safe?